core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -583,6 +583,7 @@ class SparkContext(config: SparkConf) extends Logging {
     _applicationId = _taskScheduler.applicationId()
     _applicationAttemptId = _taskScheduler.applicationAttemptId()
     _conf.set("spark.app.id", _applicationId)
+    _applicationAttemptId.foreach(attemptId => _conf.set("spark.app.attempt.id", attemptId))
```

Comment:
move conf to `conf/package.scala` ?

Reply:
IIUC conf/packages.scala includes all the public configuration keys. This should be an internal one. I am also referring to other similar codes as well.

Reply:
It seems like there are configs marked `internal` basically those won't be public to the user. For eg: `IO_WARNING_LARGEFILETHRESHOLD`, seems like `SQLConf` is honoring that couldn't find the same for regular `SparkConf`. Good to check once?

Reply:
+1 to have a conf for it and use `internal`.

Reply:
Updated with a conf in conf/package.scala